packages/framework/src/Model/Product/ProductCachedAttributesFacade.php

```diff
@@ -40,16 +35,18 @@ public function __construct(
      */
     public function getProductSellingPrice(Product $product)
     {
-        if (isset($this->sellingPricesByProductId[$product->getId()])) {
-            return $this->sellingPricesByProductId[$product->getId()];
+        $key = (string)$product->getId();
+
+        if ($this->localCacheFacade->hasItem(static::SELLING_PRICES_CACHE_NAMESPACE, $key)) {
```

Comment:
rather try-catch?

Reply:
also I am thinking, is not it possible to use `getOrSaveValue()` here?

Reply:
i wan to left some example of usage this way.